google/cloud/pubsub/internal/backwards_compatibility_type.h

```diff
@@ -0,0 +1,45 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_BACKWARDS_COMPATIBILITY_TYPE_H
+#define GOOGLE_CLOUD_CPP_GOOGLE_CLOUD_PUBSUB_INTERNAL_BACKWARDS_COMPATIBILITY_TYPE_H
+
+#include "google/cloud/pubsub/version.h"
+#include "google/cloud/options.h"
+
+namespace google {
+namespace cloud {
+namespace pubsub_internal {
+inline namespace GOOGLE_CLOUD_CPP_PUBSUB_NS {
+
+/**
+ * This is a non-constructible type. It is used to disambiguate overloaded
+ * methods that are passed an empty initializer list.
+ *
+ * See `MakeTopicAdminConnection()` for an example use case.
+ */
+struct BackwardsCompatibilityTypePleaseIgnore {
```

Comment:
I'm not sure that "backwards compatibility" is the salient feature of this type, and therefore that there might be a better name.

Reply:
Do we like `NonConstructibleTypePleaseIgnore`?